test/UnionTest.cpp

```diff
@@ -405,6 +407,75 @@ TEST(Union, cacheKeyDiffersForDifferentOrdering) {
   }
 }
 
+// _____________________________________________________________________________
+TEST(Union, cacheKeyPreventsAmbiguity) {
+  using Var = Variable;
+  auto* qec = ad_utility::testing::getQec();
+
+  auto leftT = ad_utility::makeExecutionTree<ValuesForTesting>(
+      qec, makeIdTableFromVector({{1, 4}}), Vars{Var{"?a"}, Var{"?b"}});
+
+  auto rightT = ad_utility::makeExecutionTree<ValuesForTesting>(
+      qec, makeIdTableFromVector({{1, 4}}), Vars{Var{"?a"}, Var{"?b"}});
+
+  auto bind = ad_utility::makeExecutionTree<Bind>(
+      qec, leftT->clone(),
+      parsedQuery::Bind{sparqlExpression::SparqlExpressionPimpl{
+                            std::make_unique<sparqlExpression::IdExpression>(
+                                Id::makeFromInt(1)),
+                            "1 AS ?x"},
+                        Variable{"?x"}});
+
+  Union unionOperation1{qec, std::move(bind), rightT};
+  Union unionOperation2{qec, std::move(leftT), std::move(rightT)};
+
+  // If the cache key of `unionOperation2` is a suffix of the cache key of
+  // `unionOperation1` then it is ambiguous and the bind might as well be
+  // applied to the UNION, it would be ambiguous.
```

Comment:
Please rewrite the comment such that the grammar fits the code that follows. The code is an `EXPECT_THAT` and the comment has the structure `If ... then`. That requires significant guess work regarding which aspect of the code the comment is commenting on.